crates/ra_hir_expand/src/lib.rs

```diff
@@ -157,6 +157,13 @@ impl HirFileId {
             }
         }
     }
+
+    pub fn macro_file(self) -> Option<MacroFile> {
```

Comment:
Haha, this method is so useful that every time I was working on macro related PR I urged to implement it. However, my gut feeling is that `MacroFile` should be private to `hir_expand` (right now it is public because of the limitation of salsa). And secondly, after `MacroFile` becoming to `HirFile`, we should treat it as a file as possible, and adding this method might make me forgetting this point.

But it is just my intuition, and I am happy to have it if make it `pub(crate)` :)

Reply:
Yeah, by moving the hypothetical expansion to hir_expand we can avoid this.